pkg/build/api/v1/types.go

```diff
@@ -626,6 +691,17 @@ type GenericWebHookEvent struct {
 type GitInfo struct {
 	GitBuildSource    `json:",inline"`
 	GitSourceRevision `json:",inline"`
+
+	// Refs is a list of GitRefs for the provided repo - generally sent
+	// when used from a post-receive hook. This field is optional and is
+	// used when sending multiple refs
+	Refs []GitRefInfo `json:",omitempty"`
```

Comment:
The Refs array was added for purely internal use, which is why it only exists on the internal version of the api.

it sounds like you're now trying to use the Refs field in places where we are using the external api.  Personally i don't like the way this object is defined, i think the internal version was done the way it was to preserve backwards compatibility, but i'd prefer to avoid extending the pattern into our external api because it's quite ugly.  For your usage, you should define an array of GitInfo and basically ignore the Refs field.  Can you show me where you're trying to use it as is and what error you're getting?